inception-doc/src/main/resources/META-INF/asciidoc/developer-guide/release.adoc

```diff
@@ -65,7 +65,8 @@ Do this when you want to do a feature release.
 
 == Running the release process
 
-. Check that the release branch depends on the right webanno release or update it in the most high-level `pom.xml` (properties -> `webanno.version`).
+. Check that the release branch depends on the right WebAnno release or update it in the top-level `pom.xml` (properties -> `webanno.version`).
+. Check that the JDK version you are using corresponds to the minimum Java version specified in the system requirements section of the documentation. Building against a newer version might create a release that will not run on the minimum required version because usages of APIs not available in the minimum version may not be detected.
```

Comment:
Might add how to check with `mvn -version` and how to change this (temporarily) with `export JAVA_HOME="/path/to/jvm"`